examples/example6.go

```diff
@@ -0,0 +1,112 @@
+package main
+
+import (
+	"fmt"
+	"math/rand"
+	"net/http"
+	"time"
+
+	"github.com/seborama/govcr"
+)
+
+const example6CassetteName = "MyCassette6"
+
+// Example6 is an example use of govcr.
+// This will show how to do conditional rewrites.
+// For example, your request has a "/order/{random}" path
+// and we want to rewrite it to /order/1234 so we can match it later.
+// We change the response status code.
+// We add headers based on request method.
+func Example6() {
+	cfg := govcr.VCRConfig{
+		Logging: true,
+	}
+
+	// The filter will neutralize a value in the URL.
+	// In this case we rewrite /order/{random} to /order/1234
+	replacePath := govcr.RequestFilter(func(req govcr.Request) govcr.Request {
+		// Replace path with a predictable one.
+		req.URL.Path = "/order/1234"
+		return req
+	})
+	// Only execute when we match path.
+	replacePath = replacePath.OnPath(`example\.com\/order\/`)
+
+	// Add to request filters.
+	cfg.RequestFilters.Add(replacePath)
+	cfg.RequestFilters.Add(govcr.RequestDeleteHeaderKeys("X-Transaction-Id"))
+
+	// Add filters
+	cfg.ResponseFilters.Add(
+		// Always transfer 'X-Transaction-Id' as in example 5.
+		govcr.ResponseTransferHeaderKeys("X-Transaction-Id"),
+
+		// Change status 404 to 202.
+		func(resp govcr.Response) govcr.Response {
+			if resp.StatusCode == http.StatusNotFound {
+				resp.StatusCode = http.StatusAccepted
+			}
+			return resp
+		},
+
+		// Add header if method was "GET"
+		govcr.ResponseFilter(func(resp govcr.Response) govcr.Response {
+			resp.Header.Add("method-was-get", "true")
+			return resp
+		}).OnMethod(http.MethodGet),
+
+		// Add header if method was "POST"
+		govcr.ResponseFilter(func(resp govcr.Response) govcr.Response {
+			resp.Header.Add("method-was-post", "true")
```

Comment:
best to use a different value such as `present` or `added` etc
when using 'true', this is converted to a boolean by the http client which then cannot be displayed by Println().